main.go

```diff
@@ -34,185 +24,67 @@ var (
-	tag := f.Tag.Value
-	// A field is optional if the json tag includes `omitempty`
-	name, isOptional := GetTagInfo(tag)
-	// A field is nullable if the identifier is a pointer (nil pointer --> null JSON)
-	isNullable := isNullable(f)
-	if name == "" {
-		return
-	}
-
-	fmt.Printf("  %s", name)
-	if isOptional {
-		// https://flow.org/en/docs/types/primitives/#toc-optional-object-properties
-		fmt.Printf("?: ")
-	} else if isNullable {
-		// If a type is optional AND nullable, it will not show up in the json
-		// response, so we can assume the types here are required
-		// https://flow.org/en/docs/types/primitives/#toc-maybe-types
-		fmt.Printf(": ?")
-	} else {
-		fmt.Printf(": ")
-	}
-
-	fieldType := GetTypeInfo(f.Type)
-	fmt.Print(fieldType)
-	fmt.Printf(",\n")
-}
-
-func handleTypeDef(ts ast.TypeSpec) {
-	if !ts.Name.IsExported() {
-		// Do not handle unexported structs
-		return
-	}
-
-	switch t := ts.Type.(type) {
-	// type MyAlias string
-	// type MyAlias2 AnotherType
-	case *ast.Ident:
-		fmt.Printf("type %s = %s;\n\n", ts.Name, GetTypeInfo(t))
-		return
-	// type MyAlias []AnotherType
-	case *ast.ArrayType:
-		elementType := GetTypeInfo(t.Elt)
-		fmt.Printf("type %s = Array<%s>;\n\n", ts.Name, elementType)
-		return
-	// type MyAlias map[boolean]AnotherType
-	case *ast.MapType:
-		keyType := GetTypeInfo(t.Key)
-		valueType := GetTypeInfo(t.Value)
-		fmt.Printf("type %s = {[%s]: %s};\n\n", ts.Name, keyType, valueType)
-		return
-	case *ast.StructType:
-		fmt.Printf("type %s {\n", ts.Name)
-		fields := t.Fields.List
-		for _, field := range fields {
-			handleField(*field)
-		}
-		fmt.Printf("}\n\n")
-		return
-		// Don't handle anything else
-	}
-	return
-}
-
 // TypeDefInspector handles ast nodes if they are type definitions
 func TypeDefInspector(node ast.Node) bool {
 	// Check if this node is a type definition
 	ts, ok := node.(*ast.TypeSpec)
 	if ok {
-		handleTypeDef(*ts)
+		handlers.HandleTypeDef(*ts)
 	}
 	return true
 }
 
+func parse(file string) (*ast.File, error) {
+	fset := token.NewFileSet()
+	// Parse the src file's information into the astNode, including the comments
+	astNode, err := parser.ParseFile(fset, file, nil, parser.ParseComments)
+	if err != nil {
+		return nil, err
+	}
+
+	return astNode, nil
+}
+
 func run(c *cli.Context) error {
 	file := c.String("file")
+	dir := c.String("dir")
 
 	// TODO Maxime 11/5/2017
 	// Check if the file passed in the CLI has the .go extension
-	if file == "" {
+	if file == "" && dir == "" {
 		fmt.Println("Please specify a .go file to consume")
 		return nil
 	}
-	// Create a new set of source files
-	fset := token.NewFileSet()
-	// Parse the src file's information into the astNode, including the comments
-	astNode, err := parser.ParseFile(fset, file, nil, parser.ParseComments)
+
+	// Handle directory
+	if dir != "" {
+		files, err := ioutil.ReadDir(dir)
+		if err != nil {
+			return err
+		}
+
+		for _, f := range files {
+			astNode, err := parse(dir + "/" + f.Name())
```

Comment:
can we move the `inspect` into the `parse` fn and maybe rename to `handleFile` or something like that?

then we could do something more like:

```
for _, f := range files {
    err := handleFile(dir + "/" + f.Name())
    if err != nil {
        return err
    }
}

// later in place of lines 85 -92
return handleFile(file)